### PR TITLE
feat: Add sentry cli path as parameter for actions

### PR DESCRIPTION
**Pull request by chuganzy**
This PR aims to finalize https://github.com/getsentry/sentry-fastlane-plugin/pull/42 to solve https://github.com/getsentry/sentry-fastlane-plugin/issues/34

This branch is from #42, but instead of checking cli path in individual actions, moved the logic to helper.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Please add an entry to the changelog:

```
## Unreleased

Added sentry cli path as parameter for actions [Finalize PR #42] ([#97](getsentry/sentry-fastlane-plugin/pull/97))
```

**Comment by nekrich** [comment]
Hey.
Any chances for this PR to be merged in the nearest future?

File: lib/fastlane/plugin/sentry/actions/sentry_upload_dsym.rb [hunk 1]
```diff
@@ -72,7 +71,7 @@ def self.available_options
                                       optional: true,
                                       verify_block: proc do |value|
                                         UI.user_error! "Could not find Info.plist at path '#{value}'" unless File.exist?(value)
-                                      end)
+                                      end),
```
[the remark this answers is not in the document]
Comment (denrase) [reply lib/fastlane/plugin/sentry/actions/sentry_upload_dsym.rb:74]: Yeah, should be separate. I created an issue to discuss this further. 

https://github.com/getsentry/sentry-fastlane-plugin/issues/105

**Review by denrase — COMMENTED** [review]
Alright, I was finally able to test this. I have a system where I don't have `sentry-cli` installed yet and provided a path to a binary which I downloaded from the GitHub repo. Used this PR as a local fastlane plugin. Worked as expected, good job. 👍

Screenshot 2021-11-03 at 16 25 35

Would you be so kind as to resolve the remaining conflict so we can run CI once more?

**Comment by chuganzy** [comment]
@denrase Glad to see it. Resolved the conflict! Let me know if I need anything more.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Features
+
+- Added sentry cli path as parameter for actions [Finalize PR #42] ([#97](getsentry/sentry-fastlane-plugin/pull/97))
```
Comment (denrase) [inline CHANGELOG.md:7]:
```suggestion
- Added sentry cli path as parameter for actions [Finalize PR #42](#97)
```

Comment (denrase) [reply CHANGELOG.md:7]: Seems that Danger does not like the current format. This is what's suggested.

**Review by denrase — APPROVED** [review]
LGTM! 👍 

Only change log remains and if @philipphofmann wants to add some cents.

File: CHANGELOG.md (hunk 2 above)
Comment (chuganzy) [reply CHANGELOG.md:7]: Ah, I see, makes sense.
https://github.com/getsentry/sentry-fastlane-plugin/blob/master/dangerfile.js#L2

Let me add a space before `(`!

**Comment by chuganzy** [comment]
@denrase Sorry for pinning! Is there anything else needed from my end? It would be great if we could merge this sometime soon👍

**Comment by denrase** [comment]
@chuganzy Hey! Looking good from my side, just wanted to leave this open to give the other contributors a change to leave feedback. /cc @philipphofmann

**Comment by chuganzy** [comment]
@denrase Sounds good! Thanks for the update!

**Review by philipphofmann — COMMENTED** [review]
Thanks for doing this, @chuganzy. This is the first quick pass.

File: lib/fastlane/plugin/sentry/helper/sentry_helper.rb [hunk 3]
```diff
@@ -1,39 +1,50 @@
...
+            UI.command("brew install getsentry/tools/sentry-cli")
+            UI.error("OR")
+            UI.command("curl -sL https://sentry.io/get-cli/ | bash")
+            UI.error("If you don't have homebrew, visit http://brew.sh")
+            UI.user_error!("Install sentry-cli and start your lane again!")
+          end
         end
 
-        sentry_cli_version = Gem::Version.new(`sentry-cli --version`.scan(/(?:\d+\.?){3}/).first)
+        sentry_cli_version = Gem::Version.new(`#{sentry_path} --version`.scan(/(?:\d+\.?){3}/).first)
```
[the remark this answers is not in the document]
Comment (chuganzy) [reply lib/fastlane/plugin/sentry/helper/sentry_helper.rb:21]: Makes sense, let me address it in the next push!

[the remark this answers is not in the document]
Comment (chuganzy) [reply lib/fastlane/plugin/sentry/helper/sentry_helper.rb:18]: Addressed this by adding `verify_block`!

**Comment by nekrich** [comment]
Hey there.
It's almost 6 months old, and I think @chuganzy is tired of updating his branch, asking for another review, and dealing with another reviewer's comments. We use it since October and it works great.
How can we speed up a merge process a little bit @philipphofmann , @denrase ?

**Comment by chuganzy** [comment]
Hi👋 I am still here and can make updates as needed!😀
We also started to use my branch internally, and it is one of the reasons I am now less active on this PR! It is working fine so far without any issue!

But I agree with @nekrich in regards to the faster merge process, as I see some people waiting for this feature to come! Please let me know if there is anything I can help!

**Comment by bruno-garcia** [comment]
Apologies for the slow response. @philipphofmann is away on a long break and I was supposed to take care of this repo and I dropped the ball.

Updating the branch with master, let's see if CI goes through

**Comment by philipphofmann** [comment]
Thanks, @bruno-garcia, and my apologies for not replying for a very long time @chuganzy.

**Comment by philipphofmann** [comment]
Answers to your questions:

- No, this is not available yet. We still have to do a release. I'm going to do one on Monday.
- I opened a PR to update the readme https://github.com/getsentry/sentry-fastlane-plugin/pull/124
- Yes you can also use this for sentry_check_cli_installed.

**Pull request merged**